src/InputContainer.test.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+
+import { useDispatch } from 'react-redux';
+import InputContainer from './InputContainer';
+import { ADD_RESTAURANT } from './actions';
+
+jest.mock('react-redux');
+
+describe('InputContainer', () => {
+  const dispatch = jest.fn();
+  beforeEach(() => {
+    useDispatch.mockImplementationOnce(() => dispatch);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
```

Comment:
```suggestion
  beforeEach(() => {
    jest.clearAllMocks();
    
    useDispatch.mockImplementationOnce(() => dispatch);
  });
```

이렇게 beforeEach에서 처리해주면 어떨까요? 

개인적으로는 테스트를 시작하기 전에 기존에 mocking된 것들을 클리어해주는게 좋아보이기도 하고, 매 테스트의 시작과 종료 이후에 어떤일이 발생할지를 염두해두는 것 보다, 이 경우엔 시작전에 어떤 처리를 해야할지 만 알아도 될 것 같아서요 ㅎㅎ